libraries/Microsoft.Bot.Builder/DictionaryExtensions.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Licensed under the MIT License.
+// Copyright (c) Microsoft Corporation. All rights reserved.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Builder
+{
+    /// <summary>
+    /// Extension method on object <see cref="object"/>.
+    /// </summary>
+    public static partial class DictionaryExtensions
```

Comment:
if we make this class internal and drop the changes to TestBot we are probably good to go in this - assuming unit test coverage.

Reply:
I don't think we should 'drop the changes to TestBot', as we are currently testing both None and All type name handling in TestBot tests.  If we make this internal, we can just duplicate what it is doing within TestBot (ensuring TestBot is still tested with both None and All).

Reply:
I agree that we may not want to make this part of the SDK, Don't we have other extensions that can covert JSON to typed objects somewhere else?